src/spikeinterface/extractors/phykilosortextractors.py

```diff
@@ -302,3 +315,177 @@ def __init__(self, folder_path: Path | str, keep_good_only: bool = False, remove
+    duration = sorting._sorting_segments[0]._all_spikes[-1] / sampling_frequency + 1
+
+    if (phy_path / "probe.prb").is_file():
+        probegroup = read_prb(phy_path / "probe.prb")
+        probe = probegroup.probes[0]
+    elif (phy_path / "channel_positions.npy").is_file():
+        probe = Probe(si_units="um")
+        channel_positions = np.load(phy_path / "channel_positions.npy")
+        probe.set_contacts(channel_positions)
+        probe.set_device_channel_indices(range(probe.get_contact_count()))
+    else:
+        AssertionError("Cannot read probe layout from folder {phy_path}.")
+
+    # to make the initial analyzer, we'll use a fake recording and set it to None later
+    recording, _ = generate_ground_truth_recording(
+        probe=probe, sampling_frequency=sampling_frequency, durations=[duration]
+    )
+
+    sparsity = _make_sparsity_from_templates(sorting, recording, phy_path)
+
+    sorting_analyzer = create_sorting_analyzer(sorting, recording, sparse=True, sparsity=sparsity)
+
+    # first compute random spikes. These do nothing, but are needed for si-gui to run
+    sorting_analyzer.compute("random_spikes")
+
+    _make_templates(sorting_analyzer, phy_path, sparsity.mask, sampling_frequency, unwhiten=unwhiten)
+    _make_locations(sorting_analyzer, phy_path)
+    _make_amplitudes(sorting_analyzer, phy_path)
+
+    if compute_extras:
+        sorting_analyzer.compute(
+            {
+                "unit_locations": {},
+                "correlograms": {},
+                "template_similarity": {},
+                "isi_histograms": {},
+                "template_metrics": {"include_multi_channel_metrics": True},
+                "quality_metrics": {},
+            }
+        )
+
+    sorting_analyzer._recording = None
+    return sorting_analyzer
+
+
+def _make_amplitudes(sa, phy_path: Path):
+
+    amplitudes_extension = ComputeSpikeAmplitudes(sa)
+
+    amps_np = np.load(phy_path / "amplitudes.npy")
+
+    amplitudes_extension.data = {}
+    amplitudes_extension.data["amplitudes"] = amps_np
+
+    params = {"peak_sign": "neg"}
+    amplitudes_extension.params = params
+
+    amplitudes_extension.run_info = {"run_completed": True}
+
+    sa.extensions["spike_amplitudes"] = amplitudes_extension
+
+
+def _make_locations(sa, phy_path):
+
+    locations_extension = ComputeSpikeLocations(sa)
+
+    locs_np = np.load(phy_path / "spike_positions.npy")
+
+    num_dims = len(locs_np[0])
+    column_names = ["x", "y", "z"][:num_dims]
+    dtype = [(name, locs_np.dtype) for name in column_names]
+
+    structured_array = np.array(np.zeros(len(locs_np)), dtype=dtype)
+    for a, column_name in enumerate(column_names):
+        structured_array[column_name] = locs_np[:, a]
+
+    locations_extension.data = {}
+    locations_extension.data["spike_locations"] = structured_array
+
+    params = {}
+    locations_extension.params = params
+
+    locations_extension.run_info = {"run_completed": True}
```

Comment:
same on dict assignment form, but ofc this is completely arbitrary and can be ignored